api/src/main/java/org/apache/cloudstack/api/command/admin/storage/SeedSystemVMTemplateCmd.java

```diff
@@ -0,0 +1,118 @@
+authorized = {RoleType.Admin})
+public class SeedSystemVMTemplateCmd extends BaseCmd {
+
+    @Parameter(name = ApiConstants.HYPERVISOR, type = CommandType.STRING, required = true, description = "The target hypervisor for the template.")
+    private String hypervisor;
+
+    @Parameter(name = ApiConstants.IMAGE_STORE_UUID, type = CommandType.STRING, required = true, description = "Destination image store id.")
+    private String imageStoreId;
+
+    @Parameter(name = ApiConstants.FILE_UUID, type = CommandType.STRING, required = true, description = "File uuid.")
+    private String fileUUID;
+
+    @Parameter(name = ApiConstants.TEMPLATE_ID, type = CommandType.LONG, required = false, description = "template id")
+    private Long templateId;
+
+    @Override
+    public void execute(){
+        ListResponse<ImageStoreResponse> imageStoreResponses = _queryService.searchForImageStores(new ListImageStoresCmd());
+
+        ImageStoreResponse imageStoreResponse = new ImageStoreResponse();
+
+        if (imageStoreResponses.getCount() > 1){
+            // get image store with correct id;
+            for (ImageStoreResponse response: imageStoreResponses.getResponses()){
+                if (response.getId().equalsIgnoreCase(imageStoreId)){
+                    imageStoreResponse = response;
+                }
+            }
+        } else if (imageStoreResponses.getCount() == 1) {
+            imageStoreResponse = imageStoreResponses.getResponses().get(0);
+        } else if (imageStoreResponses.getCount() == 0) {
+            throw new CloudRuntimeException("Unable to find imagestore.");
+        }
+
+        hypervisor = hypervisor.toLowerCase();
+
+        String mountPoint = "/tmp/nfsmount";
+        String uploadPath = "/tmp/upload";
+
+        try {
+            URI uri = new URI(imageStoreResponse.getUrl());
+
+            Script.runSimpleBashScript("mkdir " + mountPoint);
+            Script.runSimpleBashScript("sudo mount -t nfs " + uri.getHost() + ":" + uri.getPath() + " " + mountPoint);
+
+        } catch (URISyntaxException e) {
+            throw new CloudRuntimeException("Malformed URI " + imageStoreResponse.getUrl());
+        }
+
+        String script = Script.findScript("scripts/storage/secondary/","cloud-install-sys-tmplt");
+
+        String command = script + " -h " + hypervisor + " -F -m " + mountPoint + " -f " + uploadPath + "/" + fileUUID;
+
+        if (templateId != null){
+            command += " -t " + templateId;
+        }
+
+        Script.runSimpleBashScriptForExitValue(command);
+
+        Script.runSimpleBashScript("sudo umount " + mountPoint);
+
+        _templateService.updateTemplate(imageStoreResponse.getZoneId());
+
+        SeedSystemVMTemplateResponse response = new SeedSystemVMTemplateResponse();
+        response.setResponseName(getCommandName());
+
+        this.setResponseObject(response);
+    }
+
+    @Override
+    public String getCommandName() {
+        return "seedsystemvmtemplateresponse";
```

Comment:
Refer to hackerbook and other APIs, on how to set the name. Use static API name.